syntax = "proto3";

package session;

message SessionMsg {
     oneof inner {
         NewSessionMsg new_session = 1;
         NewDetectionMsg new_detection = 2;
         UpdateDetectionMetaMsg  update_detection_meta = 3;
         UpdateDetectionMsg update_detection = 4;
         FrameMsg stream_frame = 5;
     }
}

message NewSessionMsg {
    string session = 1;
}

message NewDetectionMsg {
    int32 detection = 1;
    int64 created = 2;
    float score = 3;
    int32 clazz = 4;
    int32 width = 5;
    int32 height = 6;
    bytes img_data = 7;
}

message UpdateDetectionMetaMsg {
    int32 detection = 1;
    int64 updated = 2;
}

message UpdateDetectionMsg {
    int32 detection = 1;
    int64 updated = 2;
    float score = 3;
    int32 width = 4;
    int32 height = 5;
    bytes img_data = 6;
}

message FrameMsg {
    int64 timestamp = 1;
    bytes frame = 2;
}

